ui/src/views/AutogenView.vue

```diff
@@ -972,6 +972,9 @@ export default {
             break
           }
         }
+        if ((action.icon === 'delete' || ['archiveEvents', 'archiveAlerts', 'unmanageVirtualMachine'].includes(action.api)) && this.dataView) {
+          this.$router.go(-1)
```

Comment:
@davidjumani is it possible to check if previous page even exist? Like I can login with a VM details page redirect and then delete that VM. In that case will this take me back to VM list?

Reply:
Will redirect to the dashboard. Right now vue doesn't support getting the history list so instead we need to use hooks for every navigation (to ensure we go up instead of to the dashboard) which will slow down the UI